test: flaky TestDASer_stopsAfter_BEFP

Failed on main
```
=== RUN   TestDASer_stopsAfter_BEFP
2022-07-18T20:20:29.366Z	ERROR	share	share/full_availability.go:62	availability validation failed	{"root": "45zl/3Xr/Ssybz1uKOqgXrDww5ox/GSyUK4F5h9l7aQ=", "err": "byzantine error. Axis:1, Index:0"}
2022-07-18T20:20:29.366Z	ERROR	das	das/daser.go:275	sampling failed	{"height": 1, "hash": "60AF2EB2227EF10770CF65484544FF9E64F43FB5188B699AA18359CC8642E129", "square width": 4, "data root": "45zl/3Xr/Ssybz1uKOqgXrDww5ox/GSyUK4F5h9l7aQ=", "err": "byzantine error. Axis:1, Index:0"}
2022-07-18T20:20:29.366Z	ERROR	das	das/daser.go:210	catch-up routine failed	{"attempted range: from": 0, "to": 15}
2022-07-18T20:23:24.253Z	ERROR	fraud	fraud/subscription.go:52	reading next proof failed	{"err": "context deadline exceeded"}
    daser_test.go:290: context deadline exceeded
--- FAIL: TestDASer_stopsAfter_BEFP (180.00s)
```

--- a/das/daser_test.go
+++ b/das/daser_test.go
@@ -255,7 +255,7 @@ func TestDASer_catchUp_fails(t *testing.T) {
 }
 
 func TestDASer_stopsAfter_BEFP(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*3)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
 	t.Cleanup(cancel)
 
 	ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
@@ -274,20 +274,23 @@ func TestDASer_stopsAfter_BEFP(t *testing.T) {
 	// create fraud service and break one header
 	f := fraud.NewService(ps, mockGet.GetByHeight, ds)
 	mockGet.headers[1] = header.CreateFraudExtHeader(t, mockGet.headers[1], bServ)
-
+	newCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
 	// create and start DASer
 	daser := NewDASer(shareServ, sub, mockGet, ds, f)
-
-	require.NoError(t, daser.Start(ctx))
-	fraud.OnProof(ctx, f, fraud.BadEncoding,
+	ch := make(chan struct{})
+	go fraud.OnProof(newCtx, f, fraud.BadEncoding,
 		func(proof fraud.Proof) {
-			require.NoError(t, daser.Stop(ctx))
+			ch <- struct{}{}
 		})
-	// wait for dasing catch-up routine fails
+	time.Sleep(time.Millisecond * 100)
+	require.NoError(t, daser.Start(newCtx))
+	// wait for fraud proof will be handled
 	select {
 	case <-ctx.Done():
 		t.Fatal(ctx.Err())
-	case <-daser.ctx.Done():
+	case <-ch:
+		require.NoError(t, daser.Stop(ctx))
 	}
 
 	require.True(t, daser.ctx.Err() == context.Canceled)

--- a/das/daser_test.go
+++ b/das/daser_test.go
@@ -274,27 +274,27 @@ func TestDASer_stopsAfter_BEFP(t *testing.T) {
 	// create fraud service and break one header
 	f := fraud.NewService(ps, mockGet.GetByHeight, ds)
 	mockGet.headers[1] = header.CreateFraudExtHeader(t, mockGet.headers[1], bServ)
-	newCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	newCtx := context.Background()
+
 	// create and start DASer
 	daser := NewDASer(shareServ, sub, mockGet, ds, f)
-	ch := make(chan struct{})
+	resultCh := make(chan error)
 	go fraud.OnProof(newCtx, f, fraud.BadEncoding,
-		func(proof fraud.Proof) {
-			ch <- struct{}{}
+		func(fraud.Proof) {
+			resultCh <- daser.Stop(ctx)
 		})
-	time.Sleep(time.Millisecond * 100)
+
 	require.NoError(t, daser.Start(newCtx))
 	// wait for fraud proof will be handled
 	select {
 	case <-ctx.Done():
 		t.Fatal(ctx.Err())
-	case <-ch:
-		require.NoError(t, daser.Stop(ctx))
+	case res := <-resultCh:
+		require.NoError(t, res)
 	}
-
 	require.True(t, daser.ctx.Err() == context.Canceled)
 }
+
 func TestDASerState(t *testing.T) {
 	ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
 	bServ := mdutils.Bserv()

